src/voice-broadcast/hooks/useHasRoomLiveVoiceBroadcast.ts

```diff
@@ -14,18 +14,26 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState } from "react";
+import { useContext, useState } from "react";
 import { Room, RoomStateEvent } from "matrix-js-sdk/src/matrix";
 
 import { hasRoomLiveVoiceBroadcast } from "../utils/hasRoomLiveVoiceBroadcast";
 import { useTypedEventEmitter } from "../../hooks/useEventEmitter";
+import { SDKContext } from "../../contexts/SDKContext";
 
 export const useHasRoomLiveVoiceBroadcast = (room: Room) => {
-    const [hasLiveVoiceBroadcast, setHasLiveVoiceBroadcast] = useState(hasRoomLiveVoiceBroadcast(room).hasBroadcast);
-
-    useTypedEventEmitter(room.currentState, RoomStateEvent.Update, () => {
-        setHasLiveVoiceBroadcast(hasRoomLiveVoiceBroadcast(room).hasBroadcast);
-    });
-
+    const sdkContext = useContext(SDKContext);
+    const [hasLiveVoiceBroadcast, setHasLiveVoiceBroadcast] = useState(false);
+
+    const update = sdkContext.client
+        ? () => {
+              hasRoomLiveVoiceBroadcast(sdkContext.client!, room).then(({ hasBroadcast }) => {
```

Comment:
There is no need here to catch the promise in case of failure ?

Reply:
There should not be any update on error